scheduler/service/service.go

```diff
@@ -819,8 +820,18 @@ func (s *Service) handleTaskFail(ctx context.Context, task *resource.Task, backT
 			for _, detail := range st.Details() {
 				switch d := detail.(type) {
 				case *errordetailsv1.SourceError:
+					var proto = "unknown"
+					if u, err := url.Parse(task.URL); err == nil {
+						proto = u.Scheme
+					}
+					// TODO currently, metrics.PeerTaskSourceErrorCounter is only updated for seed peer source error, need update for normal peer
 					if d.Metadata != nil {
 						task.Log.Infof("source error: %d/%s", d.Metadata.StatusCode, d.Metadata.Status)
+						metrics.PeerTaskSourceErrorCounter.WithLabelValues(
+							task.URLMeta.Tag, proto, fmt.Sprintf("%d", d.Metadata.StatusCode)).Inc()
+					} else {
+						task.Log.Warn("source error, but no metadata found")
+						metrics.PeerTaskSourceErrorCounter.WithLabelValues(task.URLMeta.Tag, proto, "0").Inc()
```

Comment:
Here should add a "application" label, see this pr: https://github.com/dragonflyoss/Dragonfly2/pull/1559